VMware.CloudFoundation.Reporting.psm1

```diff
@@ -824,7 +824,158 @@ Function Request-SoSHealthJson {
+    $optionsPayload | Add-Member -notepropertyname 'config' -notepropertyvalue $optionsConfigPayload
+    $optionsPayload | Add-Member -notepropertyname 'include' -notepropertyvalue $optionsIncludePayload
+    $scopeDomainsPayload = New-Object -TypeName psobject
+    $scopeDomainsPayload | Add-Member -notepropertyname 'clusterNames' -notepropertyvalue @()
+    $scopeDomainsPayload | Add-Member -notepropertyname 'domainName' -notepropertyvalue ""
+    $scopeDomainsPayload.clusterNames += @("")
+    $domainsArray = @()
+    $domainsArray += $scopeDomainsPayload
+    $scopePayload = New-Object -TypeName psobject
+    $scopePayload | Add-Member -notepropertyname 'domains' -notepropertyvalue @()
+    $scopePayload | Add-Member -notepropertyname 'includeAllDomains' -notepropertyvalue $false
+    $scopePayload | Add-Member -notepropertyname 'includeFreeHosts' -notepropertyvalue $false
+    $scopePayload.domains += $domainsArray
+    $healthSummarySpec = New-Object -TypeName psobject
+    $healthSummarySpec | Add-Member -notepropertyname 'healthChecks' -notepropertyvalue $healthChecksPayload
+    $healthSummarySpec | Add-Member -notepropertyname 'options' -notepropertyvalue $optionsPayload
+    $healthSummarySpec | Add-Member -notepropertyname 'scope' -notepropertyvalue $scopePayload
+    
+    # Request VCF Token
+    Request-VCFToken -fqdn $server -Username $user -Password $pass -skipCertificateCheck -ErrorAction SilentlyContinue -ErrorVariable ErrMsg | Out-Null
+
+    # Create VCF REST API Header
+    $vcfHeaders = @{"Content-Type" = "application/json" }
+    $vcfHeaders.Add("Authorization", "Bearer $accessToken")
+
+    Try {
+        if ($PsBoundParameters.ContainsKey("allDomains")) {
+            $healthSummarySpec.scope.includeAllDomains = $true
+            if ($PSEdition -eq "Core" -and ($PSVersionTable.OS).Split(' ')[0] -eq "Linux") {
+                $reportDestination = ($reportDestination = ($reportPath + "\" + $server.Split(".")[0] + "-all-health-results.json")).split('\') -join '/' | Split-Path -NoQualifier
+            } else {
+                $reportDestination = ($reportPath + "\" + $server.Split(".")[0] + "-all-health-results.json")
+            }
+        } elseif ($PsBoundParameters.ContainsKey("workloadDomain")) {
+            $healthSummarySpec.scope.domains[0].domainName = $workloadDomain
+            if ($PSEdition -eq "Core" -and ($PSVersionTable.OS).Split(' ')[0] -eq "Linux") {
+                $reportDestination = ($reportDestination = ($reportPath + "\" + $workloadDomain + "-all-health-results.json")).split('\') -join '/' | Split-Path -NoQualifier
+            } else {
+                $reportDestination = ($reportPath + "\" + $workloadDomain + "-all-health-results.json")
+            }
+        }
+        if (Test-VCFConnection -server $server) {
+            # Use Rest API method to request for health summary
+            $uri = "https://$server/v1/system/health-summary"
```

Comment:
Here we could instead use the `Get-VCFHealthSummaryTask` cmdlet from PowerVCF.

See: https://github.com/vmware/powershell-module-for-vmware-cloud-foundation/blob/0545475176f01a06968bcdd5d4740ada55d9c2ca/PowerVCF.psm1#L3414-L3459

Reply:
Get-VCFHealthSummaryTask is to fetch all health summary task, it does not on demand request for a health check and it is unable to specify individual workload domain.  (reference link: https://developer.vmware.com/apis/vcf/4.4.0/v1/system/health-summary/get) overall SOS procedure please see section 5. Perform Health-Check operation for SDDC (https://developer.vmware.com/apis/vcf/4.4.0/sos/)

Reply:
found something Start-VCFHealthSummary, will work on using PowerVCF cmdlets instead

Reply:
addressed in commit d64ce9ff50fe306dc3219c3013b0849ffc2e49c5